### PR TITLE
INFINITY-3409 Persistent storage changes: Namespacing, FrameworkStore, Schema check

### PR DESCRIPTION
**Pull request by nickbp**
- Support namespacing in `StateStore`/`ConfigStore`.
- Move FrameworkId storage from `StateStore` into a separate `FrameworkStore` for which namespacing does not apply.
- Remove erasing storage from `StateStore` and do it at the `Persister` level instead. Now that `StateStore` can be namespaced, erasing storage there would be confusing.
- Move schema version check upstream into the `SchedulerRunner`, instead of doing it in `StateStore`/`ConfigStore` (once each). This allows upstream to determine the expected schema version
- Clean up storage initialization in SchedulerBuilder: Just allow upstream to provide the `Persister`, and build `*Store`s based on that `Persister`.

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/state/FrameworkStore.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+     * @param persister The persister which holds the data
+     */
+    public FrameworkStore(Persister persister) {
+        this.persister = persister;
+    }
+
+    /**
+     * Returns the underlying {@link Persister}.
+     */
+    public Persister getPersister() {
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/state/FrameworkStore.java]: When would something need to access the persister?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/state/SchemaVersionStore.java [hunk 2]
```diff
@@ -65,36 +52,44 @@
      * compatible. Note that this may auto-populate the underlying schema version if the value isn't
      * currently present.
      *
-     * @return the current schema version, which may be auto-populated with the current version
+     * @param expectedVersion the expected schema version to be stored if no version is currently set or to be checked
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/state/SchemaVersionStore.java]: Java doc summary is stale

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java [hunk 3]
```diff
@@ -215,14 +216,17 @@ static void repairTaskIDs(StateStore stateStore) {
...
+     * uninstalling.
      */
     public static boolean isUninstalling(StateStore stateStore) throws StateStoreException {
         return fetchBooleanProperty(stateStore, UNINSTALLING_PROPERTY_KEY);
     }
 
     /**
-     * Sets an 'uninstall' property in the provided {@link StateStore} to {@code true}.
+     * Sets an 'uninstall' property in the provided {@link StateStore} to {@code true}. If the {@link StateStore} was
+     * created against a namespaced service, then this returns whether that service is uninstalling.
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java]: I don't see a return? Copy paste error I think...

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/state/FrameworkStore.java (hunk 1 above)
Comment (nickbp) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/state/FrameworkStore.java]: I think this (will) come up when we're clearing the persister data during uninstall, or something like that.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java (hunk 3 above)
Comment (nickbp) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java]: Fixed.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/state/SchemaVersionStore.java (hunk 2 above)
Comment (nickbp) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/state/SchemaVersionStore.java]: Updated javadocs across the class.

**Pull request merged**